hooks/namespell.py

```diff
@@ -94,8 +94,21 @@ def check_and_fix_file(filename, autofix=False):
     # Whole file ignored
     if file_rules == {}:
         return True
+    inside_codeblock = False
     for line_number, line in enumerate(lines, start=1):
         fixed_line = line
+
+        # Ignore text inside code blocks in Markdown
+        if extension == ".md":
```

Comment:
Maybe you could add

```python
class IgnoreBlock(NamedTuple):
    start: string
    end: string

IGNORE = {
".md": [IgnoreBlock("```", "```"), IgnoreBlock("<!--", "-->")]
}
```

And similar to get_comment_string add get_ignore_blocks and then

```python
ignore_blocks = get_ignore_blocks(extension)

if inside_codeblock and line.startswith(block_end):
    inside_codeblock = False
elif not inside_codeblock:
    for block in ignore_blocks:
        if line.startswith(block.start):
            inside_codeblock = True
            block_end = block.end
            break
```

That way we could ignore multiple different block types per extension.
Though it would only work if line starts with this string, script looks kinda complicated and is not to well documented so I'm not sure if it would be easy to add blocks that can start/end in middle of string.

Reply:
Done 0a39489a8331a315320582a59b97c705f7436070

It works also when the line does not start with the string. It also works with nested blocks (You can have a code blocks inside a comment block and it won't break). 
Additionally, I added verbose mode.